### PR TITLE
Get container count for docker info

### LINKED ISSUE #1389: Docker Info should get container count
Docker info should use the docker ps infrastructure to get container count info.

### PR DESCRIPTION
**Pull request by sflxn**
Use the docker ps infrastructure to retrieve container
count data for docker info.

Resolves #1389.

### CODE REVIEW

File: lib/apiservers/engine/backends/system.go [hunk 1]
```diff
@@ -262,6 +265,25 @@ func getImageCount(plClient *client.PortLayer) int {
...
 }
 
+// Use the Portlayer's support for docker ps to get the container count
 func getContainerStatus(plClient *client.PortLayer) (ContainerStatus, error) {
-	return ContainerStatus{0, 0, 0, 0}, nil
+	var status ContainerStatus
+
+	all := new(bool)
+	*all = true
+	containList, err := plClient.Containers.GetContainerList(containers.NewGetContainerListParams().WithAll(all))
```
Comment (jzt) [inline lib/apiservers/engine/backends/system.go]: Minor nit: Any reason we can't go the extra inch and call this `containerList`? :)

File: lib/apiservers/engine/backends/system.go [hunk 2]
```diff
@@ -262,6 +265,25 @@ func getImageCount(plClient *client.PortLayer) int {
...
+		return status, derr.NewErrorWithStatusCode(fmt.Errorf("Failed to get container list: %s", err), http.StatusInternalServerError)
+	}
+
+	for _, t := range containList.Payload {
+		if *t.Status == "Running" {
+			status.numRunning++
+		} else if *t.Status == "Stopped" {
+			status.numStopped++
+		}
+		status.count++
```
Comment (jzt) [inline lib/apiservers/engine/backends/system.go]: Do we care about paused containers? We'd be counting them as well here, although I'm not sure we care about them right now. `numPaused` in the `ContainerStatus` struct seems to suggest so.

**Comment by jzt** [comment]
LGTM

**Comment by hmahmood** [comment]
lgtm

File: lib/apiservers/engine/backends/system.go (hunk 2 above)
Comment (sflxn) [inline lib/apiservers/engine/backends/system.go]: yeah, it appears we're not supporting paused container for now.

File: lib/apiservers/engine/backends/system.go (hunk 1 above)
Comment (sflxn) [inline lib/apiservers/engine/backends/system.go]: lol.  I didn't even notice that.  That's so weird.  Why did I do that?  lol

**Comment by sflxn** [comment]
poke

**Pull request merged**